pkg/proc/native/threads.go

```diff
@@ -119,9 +119,14 @@ func (t *Thread) SetCurrentBreakpoint(adjustPC bool) error {
 	if err != nil {
 		return err
 	}
+	if _, ok := t.Arch().(*proc.ARM64); ok {
```

Comment:
This change here is very much not obvious and needs some kind of explanatory comment. Also I don't get it. I thought the way break instructions worked on ARM64 the value of PC would stay on the break instruction, that would make the SetPC we do later a no-op, why is this needed?

Reply:
In amd64, the PC would stay on the next instrcution of breakpoint when hitting the breakpoint. So, we need to adjust the PC back to the place of breakpoint by `SetPC()`, and then replace the breakpoint instrcution by user's origin instruction. The `FindBreakpoint()` can help us to know wheather there is hitting the breakpoint, and it will search the previous instrcution when `adjustPC` is true.
In arm64, the PC would stay on the break instruction when hitting the breakpoint. If we execute `si` command then, after single instruction executed, delve will go into this function, and `FindBreakpoint()` will return the breakpoint location(previous instrcution) and the `SetPC()` will let PC back to the breakpoint location. This will cause `si` to not step one instruction.
In arm64, we donot need to search previous instrcution, so the `adjustPC` should be always false. To the same, there is no need to execute `SetPC()` if `adjustPC` is false.

Reply:
> This change here is very much not obvious and needs some kind of explanatory comment. 

I write the comment in the head of patch https://github.com/go-delve/delve/pull/1733/commits/6f0cebdb5832eec8660b7e7b5f26913daaed1e96. I think it should be moved in sourcecode, and explain it fully。

Reply:
Ok. I get it now.

Let's add a new method to Arch:

```
BreakInstrMovesPC() bool
```

and have it return true for AMD64 and false for ARM64

then on this line do:

```
// If the breakpoint instruction does not change the value of PC after being executed we should look for breakpoints with bp.Addr == PC and there is no need to call SetPC after finding one. 
adjustPC = adjustPC && t.Arch().BreakMovesPC()
```

Reply:
modified.